src/order/antichain.lean

```diff
@@ -0,0 +1,92 @@
+/-
+Copyright (c) 2020 Bhavik Mehta, Alena Gusakov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bhavik Mehta, Alena Gusakov
+-/
+import order.basic
+import data.finset
+import data.multiset.finset_ops
+
+/-!
+# Antichains
+Investigating the structure of finsets in a partial order.
+We define antichains.
+
+## Main definitions
+* `antichain` is a finset of elements in a partial order where
+  no element is strictly less than another.
+
+* The `join` of two antichains is the set obtained by taking
+  their union and removing all elements that are less than
+  another element.
+
+* The `meet` of two antichains is their intersection.
```

Comment:
That's not what wikipedia seems to imply:

![](https://wikimedia.org/api/rest_v1/media/math/render/svg/3ebc68adce6fadba9799d27d37b7e39bc8c49e65)

Reply:
@b-mehta and I sort of went back and forth on this - we sort of agreed that the definition does end up simply being their intersection. I think that makes sense but I'm not entirely sure.

Reply:
If only you had some kind of tool to formally prove the definitions are equivalent :)

Reply:
I was told that would be redundant lol, I wanted to originally. Will do it now!

Reply:
Btw should I just credit you as an author as well at this point? I think I'm just going to switch to the `Type` version entirely

Reply:
@eric-wieser It's not what wikipedia implies by definition, but it's hard not to show they're the same definition: In particular I believe it's cleaner to use the intersection definition in Lean for much nicer definitional properties.

Reply:
I've no issue with choosing the simpler definition, but I think it would be good to include a proof that it's the same as the wikipedia one (unless wikipedia is just wrong and you have a better source, in which case updating wikipedia would be sensible!)